includes/class-analytics.php

```diff
@@ -0,0 +1,306 @@
+		$saved_disabled = get_option( self::EVENT_CATEGORY_DISABLED_OPTION, [] );
+		return ! in_array( $event_category, $saved_disabled );
+	}
+
+	/**
+	 * Get all known event categories and whether they should record events.
+	 *
+	 * @return array Array of 'name' and 'active' pairs.
+	 */
+	public static function get_event_categories() {
+		$event_categories = [];
+		foreach ( self::get_events() as $event ) {
+			$event_categories[ $event['event_category'] ] = [
+				'name'   => $event['event_category'],
+				'active' => self::event_category_enabled( $event['event_category'] ),
+			];
+		}
+		return $event_categories;
+	}
+
+	/**
+	 * Get data about all events.
+	 *
+	 * An event should contain the following fields:
+	 *    'id'             => string Event ID (e.g. "popupFormDisplayed").
+	 *    'on'             => string Type of action (e.g. "click" or "form-submit-success").
+	 *    'amp_on'         => string Type of action for AMP if different (e.g. "amp-form-submit-success").
+	 *    'element'        => string CSS selector for element (e.g. '#popup1').
+	 *    'amp_element'    => string CSS selector for AMP version of element if different.
+	 *    'event_name'     => string Name for event in GA (e.g. 'Clicked').
+	 *    'event_label'    => string Label for event in GA (e.g. 'Popup 1')
+	 *    'event_category' => string Category for event in GA (e.g. 'Newspack Announcements').
+	 * There can also be other fields specific for certain events (e.g. 'scrollSpec' for 'scroll' event listener).
+	 */
+	public static function get_events() {
+		$events = [
+			[
+				'id'             => 'socialShareClickedFacebook',
+				'on'             => 'click',
+				'element'        => 'a.share-facebook',
+				'amp_element'    => 'amp-social-share[type="facebook"]',
+				'event_name'     => 'social share',
+				'event_label'    => 'facebook',
+				'event_category' => 'NTG social',
+			],
+			[
+				'id'             => 'socialShareClickedTwitter',
+				'on'             => 'click',
+				'element'        => 'a.share-twitter',
+				'amp_element'    => 'amp-social-share[type="twitter"]',
+				'event_name'     => 'social share',
+				'event_label'    => 'twitter',
+				'event_category' => 'NTG social',
+			],
+			[
+				'id'             => 'articleRead25',
+				'on'             => 'scroll',
+				'event_name'     => get_the_title(),
+				'event_label'    => '25%',
+				'event_category' => 'NTG article milestone',
+				'scrollSpec'     => [
+					'verticalBoundaries' => [ 25 ],
+				],
+			],
+			[
+				'id'             => 'articleRead50',
+				'on'             => 'scroll',
+				'event_name'     => get_the_title(),
+				'event_label'    => '50%',
+				'event_category' => 'NTG article milestone',
+				'scrollSpec'     => [
+					'verticalBoundaries' => [ 50 ],
+				],
+			],
+			[
+				'id'             => 'articleRead100',
+				'on'             => 'scroll',
+				'event_name'     => get_the_title(),
+				'event_label'    => '100%',
```

Comment:
According to the [NTG Taxonomy](https://newsinitiative.withgoogle.com/training/states/ntg/assets/ntg-playbook.pdf#page=245): 
- the post title should be the `event_label`
- `n%` should be the `event_name` (assuming "Event Action" is `event_name`)
- there should be a `event_value` of the n percentage value (as integer)